python/hsfs/core/training_dataset_engine.py

```diff
@@ -123,6 +123,11 @@ def get_serving_vector(self, training_dataset, entry):
 
         if training_dataset.prepared_statements is None:
             self.init_prepared_statement(training_dataset)
+            # check if primary key map correspond to serving_keys. We check this during init state to reduce latency.
+            if not entry.keys() == training_dataset.serving_keys:
+                raise ValueError(
+                    "Provided primary key map doesn't correspond to serving_keys"
+                )
```

Comment:
Same comment as for Java

Reply:
set of key names, which are primary key names, has to be always the same, otherwise how application knows which feature vector to retrieve?